src/realm/object-store/sync/sync_session.cpp

```diff
@@ -1323,3 +1323,13 @@ void SyncSession::ConnectionChangeNotifier::invoke_callbacks(ConnectionState old
     }
     m_callback_index = npos;
 }
+
+util::Future<std::string> SyncSession::send_test_command(std::string body)
+{
+    util::CheckedLockGuard lk(m_state_mutex);
+    if (!m_session) {
+        return Status{ErrorCodes::RuntimeError, "Session doesn't exist to send test command on"};
```

Comment:
It doesn't look like Status() has a similar interface to Future, so this would throw an exception, since there is no `get()` in Status():
```
SyncSession::OnlyForTesting::send_test_command(*realm->sync_session(), command_request.dump()).get();
```

Reply:
a `util::Future` is implicitly constructable from a `Status`. In C++ a function can only have one return type - you can't return a `Status` or a `util::Future`, only a util::Future here - but if you return a type that your return type is trivially constructable from, the compiler will turn `return Status{...}` into `return util::Future{Status{...}}`